source/core_sntp_client.c

```diff
@@ -163,6 +163,57 @@ static uint32_t calculateElapsedTimeMs( const SntpTimestamp_t * pCurrentTime,
     return timeDiffMs;
 }
 
+/**
+ * @brief Validates the content of the SNTP context passed to the APIs to
+ * check whether it represents an initialized context.
+ *
+ * @param[in] pContext The SNTP context to validate.
+ *
+ * @return `true` if the context is validated to be initialized; `false`
+ * if any member of the context is invalid.
+ */
+static bool isContextInitialized( SntpContext_t * pContext )
+{
+    bool isValid = false;
+
+    assert( pContext != NULL );
+
+    /* Validate pointer parameters are not NULL. */
+    if( ( pContext->pTimeServers == NULL ) || ( pContext->pNetworkBuffer == NULL ) ||
+        ( pContext->resolveDnsFunc == NULL ) ||
+        ( pContext->getTimeFunc == NULL ) || ( pContext->setTimeFunc == NULL ) )
+    {
+        isValid = false;
+    }
+
+    /* Validate the size of the configured servers list, network buffer size and the state
+     * variable for the SNTP packet size.*/
+    else if( ( pContext->numOfServers == 0U ) || ( pContext->bufferSize < SNTP_PACKET_BASE_SIZE ) ||
+             ( pContext->sntpPacketSize < SNTP_PACKET_BASE_SIZE ) )
+    {
+        isValid = false;
+    }
+    /* Validate that the members of the UDP transport interface. */
+    else if( ( pContext->networkIntf.recvFrom == NULL ) || ( pContext->networkIntf.sendTo == NULL ) )
+    {
+        isValid = false;
+    }
+
+    /* If an authentication interface is provided, validate that boths its function pointer
+     * members are valid. */
+    else if( ( ( pContext->authIntf.generateClientAuth != NULL ) && ( pContext->authIntf.validateServerAuth == NULL ) ) ||
+             ( ( pContext->authIntf.generateClientAuth == NULL ) && ( pContext->authIntf.validateServerAuth != NULL ) ) )
```

Comment:
So it's okay if both of them are NULL or both of them are set? I forget the implementation details but is it necessary for only one of them to be defined? Also, the docstring for `validateServerAuth` has a minor grammatical error.

Reply:
Yes, the library requires the authentication interface to be completely defined instead of supporting only one of the functions to be defined. There is no concept of server-only authentication that NTP mentions but if the customer wants to have such kind of a functionality, they can define a `generateClientAuth` function that always returns success